code/parse/sexp.cpp

```diff
@@ -14649,58 +14683,70 @@ void multi_sexp_alter_ship_flag()
 	}
 
 	// no more data means do this to every ship in the mission
-	if (!process_data) {
+	if (!process_data)
+	{
 		alter_flag_for_all_ships(future_ships, object_flag, ship_flag, parse_obj_flag, ai_flag, set_flag);
 	}
-	else {
-		while (Current_sexp_network_packet.get_int(type)) {
-			oswpt.clear();
-			oswpt.type = type; 
+	else
+	{
+		int type;
+		while (Current_sexp_network_packet.get_int(type))
+		{
+			object_ship_wing_point_team *oswptp = nullptr;
```

Comment:
Please use `unique_ptr` here to ensure that the memory is always deallocated. If I see that correctly that should work here.

Reply:
How should that work, exactly?  The compiler didn't like it until I used `unique_ptr` in each place, like so...

```
std::unique_ptr<object_ship_wing_point_team> oswptp = nullptr;
...
oswptp = std::unique_ptr<object_ship_wing_point_team>(new object_ship_wing_point_team(shipp));
...
oswptp = std::unique_ptr<object_ship_wing_point_team>(new object_ship_wing_point_team(p_objp));
...
oswptp = std::unique_ptr<object_ship_wing_point_team>(new object_ship_wing_point_team(wingp));
...
oswptp = std::unique_ptr<object_ship_wing_point_team>(new object_ship_wing_point_team());
```

That's a lot more verbose.  And even with all that, the compiler didn't like `delete oswptp;` at the end of the block.

Reply:
After a bit of digging and a helpful link...
https://stackoverflow.com/questions/16061407/should-i-assign-or-reset-a-unique-ptr
...I think this is a bit better:

```
std::unique_ptr<object_ship_wing_point_team> oswptp;
...
oswptp.reset(new object_ship_wing_point_team(shipp));
...
oswptp.reset(new object_ship_wing_point_team(p_objp));
...
oswptp.reset(new object_ship_wing_point_team(wingp));
...
oswptp.reset(new object_ship_wing_point_team());
```

That's less verbose but still chokes on `delete`.  Although it appears that `delete` is no longer needed.

Reply:
Yes, that is how I meant that to work. `delete` is not necessary since the `unique_ptr` will do that for you once it goes out of scope which also covers cases like exceptions.
I think it is a good idea to use smart pointers wherever you can to avoid having to worry about the finer details of control flow.